### PR TITLE
Add README for configuring ARM template

### PR DESCRIPTION
**Pull request by colin-higgins**
Light documentation explaining how to add the extension to ARM Templates.

### CODE REVIEW

File: ARM/README.md [hunk 1]
```diff
@@ -0,0 +1,88 @@
+##Configuring Datadog in ARM Templates
```
Comment (andrewlock) [inline ARM/README.md:1]:
```suggestion
## Configuring Datadog in ARM Templates
```

File: ARM/README.md [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+
+ - [Parameters to accept Datadog configuration. ](https://docs.microsoft.com/en-us/azure/azure-resource-manager/templates/parameters)
+  - DD_API_KEY
+  - DD_ENV
+  - DD_SERVICE
+  - DD_VERSION
+  - DD_SITE
```
Comment (andrewlock) [inline ARM/README.md:8]: Need an extra indent for them to show properly
```suggestion
   - DD_API_KEY
   - DD_ENV
   - DD_SERVICE
   - DD_VERSION
   - DD_SITE
```

File: ARM/README.md [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+
+*To enable features such as [Runtime Metrics](https://docs.datadoghq.com/tracing/runtime_metrics/dotnet/) or [Logs Correlation](https://docs.datadoghq.com/tracing/connect_logs_and_traces/dotnet/?tab=serilog), you can specify these in your `appSettings` section, or you can pass them as parameters similar to the other values specified in the example below.*
+
+```
+{
+    ...
+    "parameters": {
+        ...
+        "dd_api_key": {
+            "defaultValue": "missing-dd-api-key-parameter",
```
Comment (andrewlock) [inline ARM/README.md:20]: I don't think we should supply default values for api key, env, service, version. Using the default values would be incorrect, so I think it makes more sense for users that they error if they don't supply them

Comment (colin-higgins) [reply ARM/README.md]: Agreed.

**Review by andrewlock — APPROVED** [review]

**Review by zacharycmontoya — APPROVED** [review]

**Pull request merged**